csrc/mmdeploy/codebase/mmpose/pose_tracker/utils.h

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#ifndef MMDEPLOY_CODEBASE_MMPOSE_POSE_TRACKER_UTILS_H
+#define MMDEPLOY_CODEBASE_MMPOSE_POSE_TRACKER_UTILS_H
+
+#include <array>
+#include <numeric>
+#include <vector>
+
+#include "mmdeploy/core/utils/formatter.h"
+#include "opencv2/core/core.hpp"
+#include "pose_tracker/common.h"
+
+namespace mmdeploy::mmpose::_pose_tracker {
+
+using std::vector;
+using Bbox = std::array<float, 4>;
+using Bboxes = vector<Bbox>;
+using Point = cv::Point2f;
+using Points = vector<cv::Point2f>;
+using Score = float;
+using Scores = vector<float>;
+
+#define POSE_TRACKER_DEBUG(...) MMDEPLOY_INFO(__VA_ARGS__)
+
+// scale = 1.5, kpt_thr = 0.3
+std::optional<Bbox> keypoints_to_bbox(const Points& keypoints, const Scores& scores, float img_h,
+                                      float img_w, float scale, float kpt_thr, int min_keypoints);
+
+// xyxy format
+float intersection_over_union(const Bbox& a, const Bbox& b);
+
+float object_keypoint_similarity(const Points& pts_a, const Bbox& box_a, const Points& pts_b,
+                                 const Bbox& box_b, const vector<float>& sigmas);
+
+template <typename T>
+void suppress_non_maximum(const vector<T>& scores, const vector<float>& similarities,
+                          vector<int>& is_valid, float thresh);
+
+inline float get_area(const Bbox& bbox) { return (bbox[2] - bbox[0]) * (bbox[3] - bbox[1]); }
+
+inline Point get_center(const Bbox& bbox) {
+  return {.5f * (bbox[0] + bbox[2]), .5f * (bbox[1] + bbox[3])};
+}
+
+inline std::array<float, 2> get_scale(const Bbox& bbox) {
+  return {bbox[2] - bbox[0], bbox[3] - bbox[1]};
+}
+
+inline Bbox get_bbox(const Point& center, const std::array<float, 2>& scale) {
+  return {
+      center.x - .5f * scale[0],
+      center.y - .5f * scale[1],
+      center.x + .5f * scale[0],
+      center.y + .5f * scale[1],
+  };
+}
+
+vector<std::tuple<int, int, float>> greedy_assignment(const vector<float>& scores,
+                                                      vector<int>& is_valid_row,
+                                                      vector<int>& is_valid_col, float thr);
+
+template <typename T>
+inline void suppress_non_maximum(const vector<T>& scores, const vector<float>& similarities,
```

Comment:
`codebase/mmdet` also implements `nms` and `iou`, any ideas to eliminate the duplicate codes?